impl/maven-impl/src/main/java/org/apache/maven/impl/DefaultPathMatcherFactory.java

```diff
@@ -52,7 +52,7 @@ public PathMatcher createPathMatcher(
             boolean useDefaultExcludes) {
         requireNonNull(baseDirectory, "baseDirectory cannot be null");
 
-        return new PathSelector(baseDirectory, includes, excludes, useDefaultExcludes);
+        return new PathSelector(baseDirectory, includes, excludes, useDefaultExcludes).simplify();
```

Comment:
Would it be possible to use a static factory method `PathSelector.of(...)` and make the constructor private to avoid having one caller forgetting to call `simplify` ?